homeassistant/components/drop_connect/coordinator.py

```diff
@@ -0,0 +1,183 @@
+"""DROP device data update coordinator object."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+from homeassistant.util.json import JSON_DECODE_EXCEPTIONS, json_loads
+
+from .const import (
+    CONF_DATA_TOPIC,
+    CONF_DEVICE_DESC,
+    CONF_DEVICE_ID,
+    CONF_DEVICE_TYPE,
+    CONF_HUB_ID,
+    DEV_HUB,
+    DOMAIN,
+    KEY_STATUS,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class DROPDeviceDataUpdateCoordinator(DataUpdateCoordinator):
+    """DROP device object."""
+
+    def __init__(self, hass: HomeAssistant, config_entry: ConfigEntry) -> None:
+        """Initialize the device."""
+        self.config_entry: ConfigEntry = config_entry
+        self._model: str = f"{config_entry.data[CONF_DEVICE_DESC]} on hub {config_entry.data[CONF_HUB_ID]}"
+        if config_entry.data[CONF_DEVICE_TYPE] == DEV_HUB:
+            self._model = f"Hub {config_entry.data[CONF_HUB_ID]}"
+        self._manufacturer: str = "Chandler Systems, Inc."
+        self._device_name: str = config_entry.data["name"]
+        self._device_information: dict[str, Any] = {}
+        super().__init__(hass, _LOGGER, name=f"{DOMAIN}-{config_entry.unique_id}")
+
+    async def DROPMessageReceived(
```

Comment:
```suggestion
    async def drop_message_received(
```

acording to python's naming conventions for [function names](https://peps.python.org/pep-0008/#function-and-variable-names)